manticore/platforms/wasm.py

```diff
@@ -0,0 +1,363 @@
+            else:
+                if isinstance(i.desc, TypeIdx):  # Imported function
+                    func_type = module.types[i.desc]
+                    if i.module == "env" and imported_version:
+                        imported_version = HostFunc(func_type, imported_version)
+                    self.store.funcs.append(
+                        imported_version
+                        if imported_version
+                        else HostFunc(
+                            func_type, partial(stub, len(func_type.result_types))  # type: ignore
+                        )
+                    )
+                    imports.append(FuncAddr(len(self.store.funcs) - 1))
+
+                elif isinstance(i.desc, TableType):
+                    self.store.tables.append(
+                        imported_version
+                        if imported_version
+                        else TableInst([None] * i.desc.limits.min, i.desc.limits.max)
+                    )
+                    imports.append(TableAddr(len(self.store.tables) - 1))
+                # Create an empty memory of the correct size and provide it as an import
+                elif isinstance(i.desc, MemoryType):
+                    self.store.mems.append(
+                        imported_version
+                        if imported_version
+                        else MemInst([0x0] * i.desc.min * 64 * 1024, i.desc.max)
+                    )
+                    imports.append(MemAddr(len(self.store.mems) - 1))
+                # Create a global and set its value to 0.
+                elif isinstance(i.desc, GlobalType):
+                    self.store.globals.append(
+                        imported_version
+                        if imported_version
+                        else GlobalInst(i.desc.value(0), i.desc.mut)
+                    )
+                    imports.append(GlobalAddr(len(self.store.globals) - 1))
+                else:
+                    raise RuntimeError(f"Don't know how to handle imports of type {type(i.desc)}")
+
+        return imports
+
+    def instantiate(
+        self,
+        env_import_dict: typing.Dict[
+            str, typing.Union[ProtoFuncInst, TableInst, MemInst, GlobalInst]
+        ],
+        supplemental_env: typing.Dict[
+            str, typing.Dict[str, typing.Union[ProtoFuncInst, TableInst, MemInst, GlobalInst]]
+        ] = {},
+        exec_start=False,
+        stub_missing=True,
+    ):
+        """
+        Prepares the underlying ModuleInstance for execution. Calls import_module under the hood, so this is probably
+        the only import-y function you ever need to call externally.
+
+        TODO: stubbed imports should be symbolic
+
+        :param env_import_dict: Dict mapping strings to functions. Functions should accept the current ConstraintSet as the first argument.
+        :param exec_start: Whether or not to automatically execute the `start` function, if it is set.
+        :return: None
+        """
```

Comment:
Not sure if there is a way to automatically check for unsynchronized docs with the params in the docs vs in the actual code, but the docs are missing `supplemental_env` and `stub_missing`

Reply:
Pycharm will do this if you go to Code -> Inspect Code. Fixed most of these, except for some cases where I didn't bother documenting `store` and `stack`